### DISCUSSION TITLE
Outline 使用 MinIO 的一些记录

### DISCUSSION THREAD

**Discussion by szc188** [outline/outline#4348] [Ideas]
1. 当 Outline 使用 https 的时候，MinIO 也必须使用 https；这是由于 Outline 直接在网页中调用 MinIO 的 API，同时浏览器限制同一网页中不能混用 http 和 https，因此 MinIO 也必须要确保能从客户端访问。
2. Outline 的 nginx proxy 必须启用 ws；因为查看浏览器请求时有用到 wss 协议。
3. AWS_S3_UPLOAD_BUCKET_URL 中不能包括 AWS_S3_UPLOAD_BUCKET_NAME 的相同字符串；否则图片相关的请求将会变成亚马逊的链接，看日志可以看得见，端口是 undefined。
4. 头像不能显示的问题是因为头像的图片文件保存到不能公共访问的 Bucket 中，浏览器通过 S3 的 API 读取时会遇到 403 权限问题，可以将该 Bucket 设置成 Public，但这会带来安全问题，应该要自定义 ACL，将其中的根路径下的 public 文件夹设置成公共读，其他文件夹依然为私有读写。
设置方法：在最新的 MinIO 网页控制台中，点击使用的 Bucket 的 Manage 按钮，然后将 Access Policy 设置为 private，然后切换到 Access Rules 选项卡，添加 Access Rules，Prefix：public/   ，Access：readonly，点击保存即可。

**Comment by dualseason** [comment]
只需要配置成如下即可，其中需要nginx进行端口转发即可，我后期直接使用了阿里云对接Outline
```
AWS_S3_FORCE_PATH_STYLE=true
AWS_REGION=zh-shenzhen
AWS_S3_UPLOAD_BUCKET_NAME=plus
AWS_S3_UPLOAD_BUCKET_URL=https://yourCompanyID.com
#AWS_S3_UPLOAD_BUCKET_URL=http://yourCompanyID.com:9000
AWS_S3_UPLOAD_MAX_SIZE=26214400
AWS_ACCESS_KEY_ID=minioadmin
AWS_SECRET_ACCESS_KEY=minioadmin
AWS_S3_ACL=private
```
